data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessorTest.java

```diff
@@ -227,6 +238,54 @@ void concludeGroup_returning_with_event_adds_event_to_records_out() {
             verify(actionHandleEventsDroppedCounter).increment(1);
             verify(actionHandleEventsOutCounter).increment(0);
             verifyNoInteractions(actionConcludeGroupEventsDroppedCounter);
+
+            verify(aggregateGroupManager).getGroupsToConclude(eq(false));
+            verify(aggregateActionSynchronizer).concludeGroup(identificationHash, aggregateGroup, false);
         }
+
+        @Test
+        void concludeGroup_after_prepare_for_shutdown() {
+            final AggregateProcessor objectUnderTest = createObjectUnderTest();
+            objectUnderTest.prepareForShutdown();
+
+            final Map.Entry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup> groupEntry = new AbstractMap.SimpleEntry<AggregateIdentificationKeysHasher.IdentificationHash, AggregateGroup>(identificationHash, aggregateGroup);
+            when(aggregateGroupManager.getGroupsToConclude(eq(true))).thenReturn(Collections.singletonList(groupEntry));
+            when(aggregateActionResponse.getEvent()).thenReturn(null);
+            when(aggregateActionSynchronizer.concludeGroup(identificationHash, aggregateGroup, true)).thenReturn(Optional.of(event));
+
+            final List<Record<Event>> recordsOut = (List<Record<Event>>) objectUnderTest.doExecute(Collections.singletonList(new Record<>(event)));
+
+            assertThat(recordsOut.size(), equalTo(1));
+            assertThat(recordsOut.get(0), notNullValue());
+            assertThat(recordsOut.get(0).getData(), equalTo(event));
+
+            verify(actionConcludeGroupEventsOutCounter).increment();
+            verify(actionHandleEventsDroppedCounter).increment(1);
+            verify(actionHandleEventsOutCounter).increment(0);
+            verifyNoInteractions(actionConcludeGroupEventsDroppedCounter);
+
+            verify(aggregateGroupManager).getGroupsToConclude(eq(true));
+            verify(aggregateActionSynchronizer).concludeGroup(identificationHash, aggregateGroup, true);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("isReadyForShutdownArgs")
+    public void isReadyForShutdown(final long groupsSize, final boolean expectedResult) {
+        when(aggregateGroupManager.getAllGroupsSize()).thenReturn(groupsSize);
+
+        final AggregateProcessor objectUnderTest = createObjectUnderTest();
+        final boolean result = objectUnderTest.isReadyForShutdown();
+        assertThat(result, equalTo(expectedResult));
+
+        verify(aggregateGroupManager).getAllGroupsSize();
+    }
+
+    private static Stream<Arguments> isReadyForShutdownArgs() {
+        return Stream.of(
+                Arguments.of(0, true),
+                Arguments.of(Math.abs(new Random().nextLong()) + 1L, false),
```

Comment:
I'd recommend having a test that specifically tests `1` since this this is the number next to a change in behavior (`0`). It can help if somebody tries to use `<`, `<=` incorrectly or some other odd behavior.

Reply:
Will add